avocado/core/result.py

```diff
@@ -96,7 +100,7 @@ def __init__(self, job):
         """
         self.job_unique_id = getattr(job, "unique_id", None)
         self.logfile = getattr(job, "logfile", None)
-        self.tests_total = getattr(job.args, 'test_result_total', 1)
+        self.tests_total = 0
```

Comment:
Not sure about the conflict (yes, there will be many as the both of you are working on runners) but I like the idea of allowing the runner to set/change the number of tests executed (it goes well with my idea of using Job API to generate and completely skip some tests, which means changing the number of tests).